docs/porting/security/psa_targets_in_mbed-os.md

```diff
@@ -0,0 +1,561 @@
+# PSA targets in Mbed OS
+
+This document describes the process of adding new PSA targets to Mbed OS, focusing on target configurations, build and validation. Mbed OS relies on [TF-M](https://git.trustedfirmware.org/trusted-firmware-m.git/tree/) for PSA Services and SPM, so the PSA target you are adding to Mbed OS **must** already be supported by TF-M.
+
+Before reading this document, please read [porting a custom board](../porting/porting-a-custom-board.html) and [porting targets](../porting/porting-targets.html), which provide step-by-step instructions to porting a new target to Mbed OS.
+
+## Glossary
+
+- **NSPE:** Non-Secure Processing Environment.
+- **PSA:** Platform Security Architecture.
+- **SPE:** Secure Processing Environment.
+- **SPM:** Secure Partition Manager.
+- **TF-M:** Trusted Firmware M (for M-class devices).
+
+## Adding new PSA targets
+
+To help create PSA targets, we have added a few generic targets to `targets/targets.json`:
+
+* `PSA_Target`: Root level PSA target.
+* `PSA_V7_M`: Single v7-M PSA generic target, which doesn't employ hardware isolation between the NSPE and the SPE. PSA secure service emulation enables PSA API compatibility.
+* `PSA_DUAL_CORE`: Dual-core PSA generic target. A dual-core PSA target has at least two cores that are either Armv7-M or Armv6-M. One core will be used for the SPE and another for the NSPE. Hardware isolation between the cores enables PSA compliance. On dual-core targets, TF-M
+runs on the SPE and provides PSA services, and Mbed OS runs on the NSPE.
+* `PSA_V8_M`: Armv8-M generic PSA target, which employs hardware to isolate the NSPE from the SPE. TF-M runs on the SPE and provides PSA services, and Mbed OS runs on the NSPE.
+<!--there are two ways of describing things in this list: chip-generic-psa and chip-psa-generic. Is that on purpose?-->
+
+### NSPE generic targets
+
+SPE binaries are generated by TF-M build tools, not Mbed-OS build tools. Therefore, when adding a new PSA target to Mbed OS, you don't need to define an SPE target; it's enough to define an NSPE target. Therefore, for dual-core (`PSA_DUAL_CORE`) and Armv8-M (`PSA_V8_M`), we have defined only PSA NSPE generic targets.
+
+### Inheritance
+
+The Mbed OS build tools have limited support for multiple inheritance. Because of this, a PSA target must inherit from the generic PSA target corresponding to the new target's architecture. The only exception is when the PSA target has to inherit from one of its family targets instead of the generic target `Target`<!--should that be `psa_target`? If so, that should be in the first sentence, not here-->.<!--when does this exception happen?--> We cover exception cases in each target section in this document.
```

Comment:
```suggestion
The Mbed OS build tools have limited support for multiple inheritance. Because of this, a PSA target must inherit from the generic PSA target corresponding to the new target's architecture. The only exception is when the PSA target has to inherit from one of its family targets (one of `PSA_V7_M`, `PSA_DUAL_CORE` and `PSA_V8_M`) instead of the generic target. We cover exception cases in each target section in this document.
```